ui/e2e/instances/details/index.spec.ts

```diff
@@ -87,3 +95,156 @@ test("the header of the instance page shows the relevant data for the workflow",
     "the button 'Open Workflow' is clickable and links to the correct URL"
   ).toHaveURL(editURL);
 });
+
+test("the diagram panel on the instance page responds to user interaction", async ({
+  page,
+}) => {
+  const newInstance = createInstance({ namespace, path: simpleWorkflowName });
+  await expect(newInstance, "wait until process was completed").toBeDefined();
+  const instanceId = (await newInstance).instance;
+  await page.goto(`/${namespace}/instances/${instanceId}`);
+
+  const diagramPanel = page.getByTestId("rf__wrapper");
+  await expect(diagramPanel, "It renders the diagram panel").toBeVisible();
+
+  const startNode = diagramPanel.getByTestId("rf__node-startNode");
+  const endNode = diagramPanel.getByTestId("rf__node-endNode");
+
+  await expect(startNode, "It renders the start node").toBeVisible();
+  await expect(endNode, "It renders the end node").toBeVisible();
+
+  const resizeButton = page.getByTestId("resizeDiagram");
+  await expect(resizeButton, "It renders the maximize button").toBeVisible();
+
+  await resizeButton.hover();
+  await expect(
+    page.getByText("maximize diagram"),
+    "it shows the correct text when hovering over the resize button"
+  ).toBeVisible();
+  // hover over something else to make the overlay disappear
+  page.getByTestId("rf__wrapper").hover();
+
+  await resizeButton.click();
+  await expect(
+    diagramPanel,
+    "It maximizes the diagram to full width of the screen"
+  ).toHaveCSS("width", "1032px");
+  await resizeButton.hover();
+
+  await expect(
+    page.getByText("minimize diagram"),
+    "it shows the correct text when hovering over the resize button"
+  ).toBeVisible();
+  // hover over something else to make the overlay disappear
+  page.getByTestId("rf__wrapper").hover();
+
+  await page.reload();
+  await expect(
```

Comment:
I'm not sure how the page size in playwright is consistent. When checking against exact pixel values here, it feels like the test can break very easy. Here is a suggestion on how we could approach it more general:

```typescript
  const minimizedWidth = (await diagramPanel.boundingBox())?.width;
  await resizeButton.click();
  const maximizedWidth = (await diagramPanel.boundingBox())?.width;
  if (minimizedWidth === undefined || maximizedWidth === undefined) {
    throw new Error("could not get width of diagram panel");
  }
  await expect(
    maximizedWidth / minimizedWidth,
    "The panel is siginificantly bigger after maximizing"
  ).toBeGreaterThan(1.5);
  // reload the page and check that the width is the exact same value than the maximizedWidth
  // ....
``` 
This way, we don't have to use exact pixel values. We could have also used `await expect(maximizedWidth).toBeGreaterThan(minimizedWidth);` in the last line, but this would also pass if the width only changes a couple of pixels.

Reply:
Great Idea 👍